File: Assets/MoveByEventName.js
#pragma strict
var eventName: String;
var direction: Vector3;
function Awake () {
	if(null != eventName) {
		var eventCenter: EventCenter = GameObject.FindWithTag('EventCenter').GetComponent(EventCenter);
		eventCenter.RegisterListener(eventCenter, eventName, gameObject, OnEventTriggered);
	}
}
var OnEventTriggered = function(e: SbiEvent) {
	var rigidbody: Rigidbody = GetComponent(Rigidbody);
	var speed: double = 10;
	var role: Role = GetComponent(Role);
	if(null != role) { speed = role.speed; }
	var velocity: Vector3 = transform.localToWorldMatrix.MultiplyVector(direction*speed);
	if((null != direction) && (null != rigidbody)) {
		rigidbody.velocity = velocity;
	}
};

File: Assets/Script/MoveByEventName.cs
using UnityEngine;
public class MoveByEventName : MonoBehaviour {
	public string eventName;
	public Vector3 direction;
	public void Awake () {
		if(null != eventName) {
			EventManager.Instance.RegisterListener(EventManager.Instance, eventName, gameObject, OnEventTriggered);
		}
	}
	public void OnEventTriggered(SbiEvent e) {
		Rigidbody rigidbody = GetComponent<Rigidbody>();
		float speed = 10;
		Role role = GetComponent<Role>();
		if(null != role) { speed = (float)role.speed; }
		Vector3 velocity = transform.localToWorldMatrix.MultiplyVector(direction*speed);
		if(null != rigidbody) {
			rigidbody.velocity = velocity;
		}
	}
}
